testing-internal/src/main/java/com/linecorp/armeria/internal/testing/TemporaryFolder.java

```diff
@@ -76,18 +70,23 @@ public void delete() throws IOException {
             return;
         }
 
-        try (Stream<Path> walk = Files.walk(root)) {
-            walk.sorted(Comparator.reverseOrder())
-                .map(Path::toFile)
-                .forEach(file -> {
-                    try {
-                        file.delete();
-                    } catch (Exception e) {
-                        logger.warn("Unexpected exception while deleting file: {}", file.getName(), e);
-                    }
-                });
+        deleteAll(root.toFile());
+        root = null;
+    }
+
+    private static void deleteAll(File file) throws IOException {
+        // Use File#listFiles() instead of Files#list() to ignore files deleted by other threads during
+        // iteration. See https://github.com/line/armeria/issues/4959
+        if (file.isDirectory()) {
+            final File[] files = file.listFiles();
```

Comment:
What would happen if someone deletes this directory right after `files.isDirectory()` returns `true`?

Reply:
`listFiles()` returns `null` if the path is invalid.
https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/io/File.java#L1175C15-L1176